demo/Views/SeekBarView.vala

```diff
@@ -19,11 +19,96 @@
  */
 
 public class SeekBarView : Gtk.Grid {
+    private Gtk.Popover preview_popover;
+    private Gtk.Label preview_label;
+
+    public SeekBarView () {
+        Object (valign: Gtk.Align.CENTER,
+                margin: 24);
+    }
+
     construct {
-        var seekbar = new Granite.Widgets.SeekBar ();
+        preview_popover = new Gtk.Popover (this);
+        preview_popover.can_focus = false;
+        preview_popover.sensitive = false;
+        preview_popover.modal = false;
+        preview_popover.valign = Gtk.Align.CENTER;
+
+        preview_label = new Gtk.Label ("");
+        preview_popover.add (preview_label);
+        preview_popover.show_all ();
+        preview_popover.set_visible (false);
+
+        var seek_bar = new Granite.Widgets.SeekBar (100);
+
+        preview_popover.relative_to = seek_bar.scale;
+
+        seek_bar.scale.motion_notify_event.connect ((event) => {
+            update_pointing ((int) event.x);
+            if (!seek_bar.is_grabbing) {
+                var duration_decimal = (event.x / ((double) event.window.get_width ()));
+                var duration_mins = Granite.DateTime.seconds_to_time ((int) (duration_decimal * seek_bar.playback_duration));
+                preview_label.label = duration_mins.to_string ();
+            }
+            return false;
+        });
+
+        seek_bar.scale.enter_notify_event.connect (() => {
+            preview_popover.set_visible (true);
+            return false;
+        });
+
+        seek_bar.scale.leave_notify_event.connect (() => {
+            preview_popover.set_visible (false);
+            return false;
+        });
+
+        seek_bar.scale.button_press_event.connect (() => {
+            preview_label.height_request = 50;
+            return false;
+        });
+
+        seek_bar.scale.button_release_event.connect (() => {
+            preview_label.height_request = 0;
+            return false;
+        });
+
+        seek_bar.scale.change_value.connect ((scroll, new_value) => {
+            if (new_value >= 0.0 && new_value <= 1.0) {
+                var duration_mins = Granite.DateTime.seconds_to_time ((int) (new_value * seek_bar.playback_duration));
+                preview_label.label = duration_mins.to_string ();
+            }
+            return false;
+        });
+
+        add (seek_bar);
+
+        int progress = 0;
+        Timeout.add (500, () => {
+            if (seek_bar.is_grabbing) return true;
```

Comment:
Should this be managed here or in `set_progress` in the SeekBar?

Reply:
Not sure what you mean be this. This bit of code is used to simulate behaviour of a progressing media player.

Reply:
Of course. I'm referring strictly to the `if (seek_bar.is_grabbing) return true;` part. Also please add braces `{}` around `return true` as per code style.

Reply:
I think it should be managed there because the SeekBar, internally, doesn't decide how it should progress (the progression is decided externally) therefore, what happens when the Scale is grabbed, is decided by the user of the SeekBar.